### PR TITLE
Reduce dependencies

### LINKED ISSUE #602: Reduce dependencies for AEMaaCS
For planned changes in AEMaaCS build process (in particular the image build) the required dependencies should be reduced, in particular the following changes should be made: 
* JSP API can be embedded
* Sling Commons Scheduler can be made optional (https://github.com/Netcentric/accesscontroltool/blob/develop/docs/ApplyConfig.md#upload-listener-service is rarely used and discouraged anyway, for up-and-running AEMaaCS instances it will still work but not for /apps and /libs)

### PR DESCRIPTION
**Pull request by ghenzler**
Closes #602

### CODE REVIEW

File: accesscontroltool-bundle/bnd.bnd [hunk 1]
```diff
@@ -13,4 +13,6 @@ com.adobe.granite.keystore;resolution:=optional,\
 com.adobe.granite.jmx.annotation;resolution:=optional,\
 com.day.cq.security.util;resolution:=optional,\
 org.bouncycastle.*;resolution:=optional,\
+org.apache.sling.commons.scheduler.*;resolution:=optional,\
```
Comment (kwin) [inline accesscontroltool-bundle/bnd.bnd:16]: we might need to reconsider optional imports: https://github.com/adobe/aem-core-wcm-components/issues/1872

**Review by kwin — CHANGES_REQUESTED** [review]
We should adjust https://github.com/Netcentric/accesscontroltool/blob/develop/target-osgi-environment/maximum-environment/pom.xml to check that it runs without those dependencies!

File: accesscontroltool-bundle/pom.xml [hunk 2]
```diff
@@ -106,7 +106,15 @@
             <artifactId>bcpkix-jdk15on</artifactId>
             <optional>true</optional>
         </dependency>
-        
+
+        <!-- JSP API to be embedded directly -->
+        <dependency>
+          <groupId>org.apache.tomcat</groupId>
+          <artifactId>jsp-api</artifactId>
+          <version>6.0.14</version>
```
Comment (kwin) [inline accesscontroltool-bundle/pom.xml]: dependency-checks considers this version vulnerable

File: accesscontroltool-bundle/bnd.bnd (hunk 1 above)
Comment (ghenzler) [reply accesscontroltool-bundle/bnd.bnd:16]: I see, also interesting the discussion on [Felix mailing list](https://lists.apache.org/thread.html/r20cceffd651904eda7e6512b2699b5381ab8044e352bb27cb14e9964%40%3Cusers.felix.apache.org%3E) - for `org.apache.sling.commons.scheduler` it should work reliably though, in AEM it starts at start level 19 (before the AC Tool is wired), in Sling it comes slightly later ([start level 20](https://github.com/apache/sling-org-apache-sling-starter/blob/5e9f0fc352d0a9ea855409d4a90283f4397f006b/src/main/features/base.json#L70), yet it should still be ok)

File: accesscontroltool-bundle/pom.xml (hunk 2 above)
Comment (ghenzler) [reply accesscontroltool-bundle/pom.xml]: this is currently using the same version as https://github.com/apache/sling-org-apache-sling-scripting-jsp-api/blob/8c36e309e7eae7277882445727733b400026de59/pom.xml#L48 - checking if I can get a better version without vulnerability

Comment (kwin) [reply accesscontroltool-bundle/pom.xml]: by embedding it we have to take care of that vulnerabilities don't affect us, if just using the provided one Adobe does that for us

Comment (ghenzler) [reply accesscontroltool-bundle/pom.xml]: 0a18191 fixes this and uses the latest dependency (the group id changed from `jsp-api` to `tomcat-jsp-api`, that wasn't immediately obvious)

**Comment by ghenzler** [comment]
> We should adjust https://github.com/Netcentric/accesscontroltool/blob/develop/target-osgi-environment/maximum-environment/pom.xml to check that it runs without those dependencies!

d30e5bf removes scripting dependencies in min environment (the max env didn't include them)

File: accesscontroltool-bundle/pom.xml (hunk 2 above)
Comment (ghenzler) [reply accesscontroltool-bundle/pom.xml]: turns out that the current version still has one vulnerability left: CVE-2020-8022

Comment (kwin) [reply accesscontroltool-bundle/pom.xml]: This is a false-positive: https://github.com/Netcentric/accesscontroltool/issues/596#issuecomment-922831860

File: accesscontroltool-bundle/bnd.bnd [hunk 3]
```diff
@@ -3,8 +3,11 @@ Bundle-SymbolicName: biz.netcentric.cq.tools.accesscontroltool.bundle
 # export all versioned packages by default
 -exportcontents: ${packages;VERSIONED}
 # Embed (https://groups.google.com/g/bndtools-users/c/NAdU0_X8OKY)
--includeresource: tomcat-el-api-[0-9.]*.jar;lib:=true,tomcat-jasper-el-[0-9.]*.jar;lib:=true,snakeyaml-[0-9.]*.jar;lib:=true
-# 
+-includeresource: tomcat-el-api-[0-9.]*.jar;lib:=true,tomcat-jasper-el-[0-9.]*.jar;lib:=true,snakeyaml-[0-9.]*.jar;lib:=true,tomcat-jsp-api-[0-9.]*.jar;lib:=true
+
+# avoid the dependency to servlet API coming in via tomcat-jsp-api
+-contract: !JavaServlet,*
```
Comment (kwin) [inline accesscontroltool-bundle/bnd.bnd]: AEM 6.5.x only provides support for Servlet Spec 3.1 while Tomcat 9 is for Servlet Spec 4.0 (https://tomcat.apache.org/whichversion.html), we should therefore stick to Tomcat 8, the require capability is there for a reason in Tomcat 9!

File: accesscontroltool-bundle/pom.xml [hunk 4]
```diff
@@ -22,9 +22,13 @@
     <packaging>jar</packaging>
     <name>Access Control Tool Bundle</name>
     <description>The OSGi bundle containing all classes of the Access Control Tool.</description>
+    
+    <properties>
+        <tomcat.libs.version>10.0.12</tomcat.libs.version>
```
Comment (kwin) [inline accesscontroltool-bundle/pom.xml]: Currently documentation states that we support EL 2.2 (https://github.com/Netcentric/accesscontroltool/blob/develop/docs/AdvancedFeatures.md#expressions). What has changed in 4.0? Is that really fully backwards compatible?

File: accesscontroltool-bundle/bnd.bnd (hunk 3 above)
Comment (ghenzler) [reply accesscontroltool-bundle/bnd.bnd]: we currently don't have the `Require-Capability` for the servlet API, the header comes from the embedded dependency (that is only embedded because fo the EL functionality)

Comment (kwin) [reply accesscontroltool-bundle/bnd.bnd]: we depend on this dependency having the contract, so transitively we also depend on the capability!

File: accesscontroltool-bundle/pom.xml (hunk 4 above)
Comment (ghenzler) [reply accesscontroltool-bundle/pom.xml]: The unit tests pass... if even, only edge cases could cause a problem. 

Also more generally, as EL is used broadly in a huge amount of systems, I would assume new versions are carefully tested to be backwards-compatible.

Comment (kwin) [reply accesscontroltool-bundle/pom.xml]: We need at least an update of the documentation that we now support EL 4.0 with the according link.

Comment (ghenzler) [reply accesscontroltool-bundle/pom.xml]: Updated doc in ba2634b

File: accesscontroltool-bundle/pom.xml (hunk 2 above)
Comment (kwin) [reply accesscontroltool-bundle/pom.xml]: No longer relevant, as JSP dependency has  been removed.

**Review by kwin — APPROVED** [review]
